### PR TITLE
Fix null check in bloom_filter Spark aggregate 

### PR DESCRIPTION
**Pull request by marin-ma**
According to the DecodedVector::mayHaveNulls definition, it's uncertain whether there are nulls when `mayHaveNulls` is true.
https://github.com/facebookincubator/velox/blob/e8e84e38444c9292c2a79e0ea5c72a5c6312482f/velox/vector/DecodedVector.h#L184-L188

In Gluten, we found with current implementation, some of the Spark TPCDS queries that uses the BloomFilter aggregate fails because the `bloom_filter_agg` incorrectly fail based on the `mayHaveNulls` flag. We should remove the original check and, if `mayHaveNulls` is true, add a check for each row individually.

### CODE REVIEW

File: velox/functions/sparksql/aggregates/BloomFilterAggAggregate.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ struct BloomFilterAccumulator {
   }
 
   BloomFilter<StlAllocator<uint64_t>> bloomFilter;
-}; // namespace
+};
```
Comment (xumingming) [inline velox/functions/sparksql/aggregates/BloomFilterAggAggregate.cpp:59]: unintentional change?

Comment (marin-ma) [reply velox/functions/sparksql/aggregates/BloomFilterAggAggregate.cpp:59]: Looks like the comment was in wrong place. It's not the boundary of the anonymous namespace.

Comment (xumingming) [reply velox/functions/sparksql/aggregates/BloomFilterAggAggregate.cpp:59]: Double checked, you are right.

**Comment by marin-ma** [comment]
@mbasmanova Could you help to review? Thanks!

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Comment by jinchengchenghh** [comment]
The input vector origins from xxhash() result, the null buffer should be nullptr, do you try 
`resultRef->clearNulls(rows);` after this line? https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/Hash.cpp#L363`

**Pull request closed without merging**